internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/integration/utils/VeniceServerWrapper.java

```diff
@@ -225,28 +229,38 @@ static StatefulServiceProvider<VeniceServerWrapper> generateService(
       // generate the kafka cluster map in config directory
       VeniceConfigLoader.storeKafkaClusterMap(configDirectory, kafkaClusterMap);
 
-      if (!forkServer) {
-        VeniceConfigLoader veniceConfigLoader =
-            VeniceConfigLoader.loadFromConfigDirectory(configDirectory.getAbsolutePath());
+      String d2ServiceName = D2TestUtils.getRouterD2ServiceName(null, clusterName);
 
+      VeniceProperties serverProperties = serverPropsBuilder.build();
+      boolean https = serverProperties.getBoolean(ROUTER_HTTP2_INBOUND_ENABLED, false);
+      String httpURI = "http://localhost:" + listenPort;
+      String httpsURI = "https://localhost:" + sslPort;
+      List<ServiceDiscoveryAnnouncer> d2Servers = new ArrayList<>();
+      String d2ClusterName = D2TestUtils.setupD2Config(zkAddress, https, d2ServiceName);
+      d2Servers.addAll(D2TestUtils.getD2ServersForRouter(zkAddress, d2ClusterName, httpURI, httpsURI));
+      String clusterDiscoveryD2ClusterName =
+          D2TestUtils.setupD2Config(zkAddress, https, CLUSTER_DISCOVERY_D2_SERVICE_NAME);
+      d2Servers.addAll(D2TestUtils.getD2ServersForRouter(zkAddress, clusterDiscoveryD2ClusterName, httpURI, httpsURI));
```

Comment:
Servers shouldn't announce to cluster discovery D2 service. I think that removing this might fix some test failures